HDT Creation Support

blschatz opened the pull request:
This patch enables creation of HDT's from turtle RDF files. 
Compiles the Serd parser in the extension.
Also includes a fix for the recent search regression.

blschatz committed:
Enable creation of HDT indexes from TURTLE RDF files.
Include Serd parder dependency in setup.

blschatz committed:
Win32 build support.

blschatz committed:
Fix regression in search.

Callidon commented:
1) What do you call "a regression in search"?
2) You pull request doesn't build on Travis, as you have not included ser installation. Please fix it.
3) Add unit tests for testing the new functionalities you added.

blschatz committed:
Add unit test covering search of a resource that does not exist in the HDT.
Add download of serd to travis configuration.

blschatz commented:
Commit faaab97 redefined search using the searchIDs() function rather than the hdt->search(). The searchID's function doesn't check the result of  stringToTripleID(), which results in missing strings matching *.  Please see the PR for the fix.

blschatz committed:
Attempt to fix travis build failing with error:

serd-0.30.0/src//serdi.c:58:3: error: ‘for’ loop initial declarations are only allowed in C99 mode
   for (const Syntax* s = syntaxes; s->name; ++s) {
   ^
error: command 'gcc' failed with exit status 1

blschatz committed:
Add test demonstrating new HDT creation API.

blschatz committed:
Try fix suspected path issue with Travis CI.

blschatz committed:
Try fix suspected path issue with Travis CI.

blschatz committed:
Add missing test file.

blschatz committed:
Remove scaffolding test. Travic CI build now works. A missing test source file was the cause.
